plugin/trino-druid/src/test/java/io/trino/plugin/druid/TestDruidCaseInsensitiveMatch.java

```diff
@@ -0,0 +1,90 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.druid;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.MaterializedResult;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.assertions.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+import static io.trino.plugin.druid.BaseDruidIntegrationSmokeTest.SELECT_FROM_ORDERS;
+import static io.trino.plugin.druid.BaseDruidIntegrationSmokeTest.SELECT_FROM_REGION;
+import static io.trino.plugin.druid.DruidQueryRunner.copyAndIngestTpchData;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true)
+public class TestDruidCaseInsensitiveMatch
+        extends AbstractTestQueryFramework
+{
+    private TestingDruidServer druidServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        druidServer = new TestingDruidServer();
+        closeAfterClass(() -> {
+            druidServer.close();
+            druidServer = null;
+        });
+        DistributedQueryRunner queryRunner = DruidQueryRunner.createDruidQueryRunnerTpch(
+                druidServer, ImmutableMap.of(), ImmutableMap.of("case-insensitive-name-matching", "true"));
+        copyAndIngestTpchData(queryRunner.execute(SELECT_FROM_ORDERS + " LIMIT 10"), this.druidServer, "orders", "CamelCase");
+        return queryRunner;
+    }
+
+    @Test
+    public void testNonLowerCaseTableName()
+    {
+        MaterializedResult expectedColumns = MaterializedResult.resultBuilder(getQueryRunner().getDefaultSession(), VARCHAR, VARCHAR, VARCHAR, VARCHAR)
+                .row("__time", "timestamp(3)", "", "")
+                .row("clerk", "varchar", "", "") // String columns are reported only as varchar
+                .row("comment", "varchar", "", "")
+                .row("custkey", "bigint", "", "") // Long columns are reported as bigint
+                .row("orderdate", "varchar", "", "")
+                .row("orderkey", "bigint", "", "")
+                .row("orderpriority", "varchar", "", "")
+                .row("orderstatus", "varchar", "", "")
+                .row("shippriority", "bigint", "", "") // Druid doesn't support int type
+                .row("totalprice", "double", "", "")
+                .build();
+        MaterializedResult actualColumns = computeActual("DESCRIBE " + "CamelCase");
+        Assert.assertEquals(actualColumns, expectedColumns);
+        MaterializedResult materializedRows = computeActual("SELECT * FROM druid.druid.CAMELCASE");
+        Assert.assertEquals(materializedRows.getRowCount(), 10);
+        MaterializedResult materializedRows1 = computeActual("SELECT * FROM druid.CamelCase");
+        MaterializedResult materializedRows2 = computeActual("SELECT * FROM druid.camelcase");
```

Comment:
Can be simplified using `assertQuery("SELECT COUNT(1) FROM druid.druid.camelcase", "VALUES 10")`